workers/data_refinery_workers/processors/illumina.py

```diff
@@ -43,31 +44,134 @@ def _prepare_files(job_context: Dict) -> Dict:
 
-    # Sanitize this file so R doesn't choke.
-    # Some have comments, some have non-comment-comments.
-    with open(original_file.absolute_file_path, "r") as file_input:
-        with open(job_context["input_file_path"], "w") as file_output:
+    return job_context
+
+
+def _detect_encoding(job_context: Dict) -> Dict:
+    """Some Illumina files are not encoded using utf-8, so we need to use
+    `file` to detect their encoding"""
+
+    try:
+        encoding = subprocess.check_output(
+            ["file", "--brief", "--mime-encoding", job_context["input_file_path"],],
+            encoding="utf-8",
+        ).strip()
+    except subprocess.CalledProcessError as e:
+        logger.exception(
+            "Failed to detect the input file's encoding",
+            processor_job=job_context["job_id"],
+            input_file=job_context["input_file_path"],
+        )
+        job_context["job"].failure_reason = "Failed to detect the input file's encoding"
+        job_context["success"] = False
+        job_context["job"].no_retry = True
+
+    if encoding not in ["us-ascii", "utf-8", "iso-8859-1"]:
+        logger.error(
+            "Input file has unrecognized encoding",
+            input_file=job_context["input_file_path"],
+            encoding=encoding,
+        )
+        job_context["job"].failure_reason = f"Input file has unrecognized encoding {encoding}"
+        job_context["success"] = False
+        job_context["job"].no_retry = True
+
+    job_context["encoding"] = encoding
+
+    return job_context
+
+
+def _sanitize_input_file(job_context: Dict) -> Dict:
+    """Remove all of the SOFT-specific extensions in the original file (see
+    https://www.ncbi.nlm.nih.gov/geo/info/soft.html) plus some extra things we
+    found that make R choke.  Also, some files aren't utf-8 encoded, so we will
+    re-encode them into utf-8."""
+
+    with open(job_context["input_file_path"], "r", encoding=job_context["encoding"]) as file_input:
+        with open(job_context["sanitized_file_path"], "w", encoding="utf-8") as file_output:
             for line in file_input:
-                if (
-                    "#" not in line
-                    and line.strip() != ""
-                    and line != "\n"
-                    and "\t" in line
-                    and line[0:3].upper() != "GSM"
-                    and line[0] != "'"
-                    and line[0] != '"'
-                    and line[0] != "!"
-                    and line[0] != "/"
-                    and line[0] != "<"
-                    and line[0] != "\t"
-                ):
+                HEADER_CHARS = ["#", "!", "^"]
+
+                # Sometimes we have a quoted header, so we need to check both
+                is_header = line[0] in HEADER_CHARS or (
+                    line[0] in ["'", '"'] and line[1] in HEADER_CHARS
+                )
+                is_empty = line.strip() == ""
```

Comment:
Would it not make sense to strip the line before any operations? That way a leading space doesn't make us misinterpret header lines?